### PR TITLE
Clean-up of Credentials Status Bar Item code and functionality

### PR DESCRIPTION
**Pull request by awschristou**
## Description

Changes related to the StatusBar Item that relates to the toolkit's active credentials.

Behavior changes:
- instead of hiding the item when there are no credentials in use, the text is updated to reflect this state
  image
- item now has a tooltip
- item can now be clicked on to change the active credentials

Code changes:
- removed unnecessary interface+class wrapping the status bar item because it is never accessed after initialization
- StatusBar item text is now populated based on the credentials changed event data

## Motivation and Context

In preparation for some new credentials handling code, some of the existing credentials related code flows are being tidied up in separate, isolated changes.

## Testing

* Change credentials, see status bar text updates
* see that status bar shows expected text when there are no active credentials
* hover over status bar, verify that tooltip shows as expected
* clicking on status bar item prompts user to select credentials

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/credentials/awsCredentialsStatusBarItem.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import * as vscode from 'vscode'
+import { AwsContext, ContextChangeEventsArgs } from '../shared/awsContext'
+
+const STATUSBAR_PRIORITY = 100
```
Comment (rli) [inline src/credentials/awsCredentialsStatusBarItem.ts]: Is there a reason why we're explicitly setting the priority?

Comment (awschristou) [reply src/credentials/awsCredentialsStatusBarItem.ts]: Probably not at this point. I preserved that from existing code.

File: package.nls.json [hunk 2]
```diff
@@ -53,6 +53,9 @@
     "AWS.command.quickStart": "View Quick Start",
     "AWS.command.quickStart.title": "AWS Toolkit - Quick Start",
     "AWS.command.quickStart.error": "There was an error retrieving the Quick Start page",
+    "AWS.credentials.statusbar.no.credentials": "(not connected)",
+    "AWS.credentials.statusbar.text": "AWS Credentials: {0}",
```
Comment (bryceitoc9) [inline package.nls.json]: Not sure if I'm a fan of this messaging. I think I'd prefer something completely different from how a set credential appears; make the whole displayed message read "Not connected to AWS" or similar.

The absolute worst case scenario is that a user has a credential profile named `(not connected)`

Comment (awschristou) [reply package.nls.json]: > make the whole displayed message read "Not connected to AWS" or similar.

My only concern here is to give it text that is still meaningful in the context of looking at VS Code as a whole. By keeping the same format, the aim is that it resonates more that it relates to the AWS Toolkit. Perhaps the tooltip helps with that though.

> The absolute worst case scenario is that a user has a credential profile named (not connected)

I put very low odds on this happening.

Comment (bryceitoc9) [reply package.nls.json]: Maybe "No AWS Credentials"? Still gets across that this is an area for AWS credentials but puts the fact that there aren't set credentials first. Not a hill I'm willing to die on, though; it just feels like slightly less mental energy expended.

Comment (awschristou) [reply package.nls.json]: This is what the JetBrains toolkit shows:

image

How do you feel about `AWS Credentials: None selected` (retaining the `AWS Credentials: {0}` format)

Comment (bryceitoc9) [reply package.nls.json]: ^ That someone is going to name their credentials "None selected" to mess with someone :P 

Again, not a hill I'm willing to die on, I think I like the parenthesis more than this if you're dead set on keeping the format.

**Pull request merged**